vectorstores/alloydb/vectorestores.go

```diff
@@ -0,0 +1,346 @@
+package alloydb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/internal/alloydbutil"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+const (
+	defaultIndexNameSuffix = "langchainvectorindex"
+	similaritySearchQuery  = `WITH filtered_embedding_dims AS MATERIALIZED (
+    SELECT
+        *
+    FROM
+        %s
+    WHERE
+        vector_dims (
+                embedding
+        ) = $1
+)
```

Comment:
Why does this query need to be materialized here?

Reply:
noticed it wasn't used, removed it!